autotest/ogr/ogr_shape.py

```diff
@@ -4479,7 +4479,30 @@ def ogr_shape_97():
     return 'success'
 
 ###############################################################################
-#
+# Test restore function when .shx file is missing
+
+def ogr_shape_98():
+    
+    if gdaltest.shape_ds is None:
+        return 'skip'
+    
+    gdal.SetConfigOption('SHAPE_RESTORE_SHX', 'TRUE')
+    shutil.copy( 'data/can_caps.shp', 'tmp/can_caps.shp' )
```

Comment:
The config option should be unset just after it is no longer used so as to avoid disturbing any future later tests